### DISCUSSION TITLE
Print only specific atoms from trajectory in ARC file

### DISCUSSION THREAD

**Discussion by jacekkozuch** [TinkerTools/tinker-gpu#205] [General]
I would be interested in writing out only specific atom positions (and their induced dipoles into the uind file). The reason is that I want to perform a 10-20 ns MS, but I am interested in just a few atom positions every 1 fs. I went through the documentation, but haven't found a specific keyword for this.
Thanks in advance!

**Comment by jayponder** [comment]
There is no way to do this directly with keywords. You can trim the full trajectory archive file after running the simulation using the ARCHIVE program, or just with a simple shell script. But in your particular case, I assume you want to avoid the large "unnecessary" storage of the whole system at each 1 fs time step.

By far the easiest solution, if you can do a small amount of programming, would be to make a custom version of the "mdsave" routine from Fortran Tinker8, so that it only writes out the coordinates and induced dipoles for the atoms you require. Then compile and link a custom Tinker9 GPU executable linked against this custom Tinker8 version. This should work correctly, as I'm fairly sure that the GPU Tinker9 is using calls to Tinker8 routines to write out trajectory information.

**Reply by jacekkozuch** [reply]
Exactly, it's for avoiding "the large "unnecessary" storage of the whole system at each 1 fs time step".

I see. I think in this case it will be easiest for me to write a python script to run my MD in chunks, process it after each is finished and get rid of the large files.

Thanks!

**Reply by zhi-wang** [reply]
Another problem with this request is that it would then require passing certain info from GPU to CPU and modifying a file on disk every time step. It will kill the performance of the simulation.

**Reply by jacekkozuch** [reply]
Ok, interesting, I didn't consider this. But that's very useful info. As said, I think it's easiest to chunk the MD.

Thanks a lot for the suggestions!
